src/wc/wc.rs

```diff
@@ -153,44 +212,79 @@ fn wc(files: Vec<String>, settings: &Settings) -> StdResult<(), i32> {
         let mut longest_line_length: usize = 0;
         let mut raw_line = Vec::new();
 
-        // reading from a TTY seems to raise a condition on, rather than return Some(0) like a file.
-        // hence the option wrapped in a result here
-        while match reader.read_until(LF, &mut raw_line) {
-            Ok(n) if n > 0 => true,
-            Err(ref e) if !raw_line.is_empty() => {
-                show_warning!("Error while reading {}: {}", path, e);
-                !raw_line.is_empty()
-            }
-            _ => false,
-        } {
-            // GNU 'wc' only counts lines that end in LF as lines
-            if *raw_line.last().unwrap() == LF {
-                line_count += 1;
+        #[cfg(unix)]
+        {
+            // In the special case where we only need to count the number of
+            // bytes. There are several optimizations we can do.
+            if only_need_to_count_bytes {
+                match fstat(handle.file_descriptor) {
+                    Ok(stat) => {
+                        // If the file is regular, then the `st_size` should hold
+                        // the file's size in bytes.
+                        if (stat.st_mode & S_IFREG) != 0 {
+                            byte_count = stat.st_size as usize;
+                        } else {
+                            #[cfg(any(target_os = "linux", target_os = "android"))]
+                            {
+                                // Else, if we're on Linux and our file is a FIFO pipe
+                                // (or stdin), we use splice to count the number of bytes.
+                                if (stat.st_mode & S_IFIFO) != 0 {
+                                    match count_bytes_using_splice(handle.file_descriptor) {
```

Comment:
This should just be:
```rust
// Try to use `splice()`, but fall back on `read()` if we fail
if let Ok(res) = count_bytes_using_splice(handle.file_descriptor) {
    byte_count = res;
}
```

Reply:
Actually, if you add the stuff mentioned above to `uucore`, this whole setup could probably be extensively refactored, as the `uucore` readers/writers should just fallback on normal reads/writes.